chromium_src/extensions/common/constants.h

```diff
@@ -24,6 +30,73 @@ inline constexpr char crl_set_extension_id[] =
 
 inline constexpr char google_translate_extension_id[] =
     "aapbdbdomjkkjkaonfhkkikfgjllcleb";
+
+namespace extensions_mv2 {
+
+inline constexpr char kNoScriptId[] = "bgkmgpgeempochogfoddiobpbhdfgkdi";
+inline constexpr char kUBlockId[] = "jcokkipkhhgiakinbnnplhkdbjbgcgpe";
+inline constexpr char kUMatrixId[] = "fplfeajmkijmaeldaknocljmmoebdgmk";
+inline constexpr char kAdGuardId[] = "ejoelgckfgogkoppbgkklbbjdkjdbmen";
+
+inline constexpr char kWebStoreNoScriptId[] =
+    "doojmbjmlfjjnbmnoijecmcbfeoakpjm";
+inline constexpr char kWebStoreUBlockId[] = "cjpalhdlnbpafiamejdnhcphjbkeiagm";
+inline constexpr char kWebStoreUMatrixId[] = "ogfcmafjalglgifnmanfmnieipoejdcf";
+inline constexpr char kWebStoreAdGuardId[] = "gfggjaccafhcbfogfkogggoepomehbjl";
+
+inline constexpr auto kBraveHosted =
+    base::MakeFixedFlatMap<std::string_view, std::string_view>(
+        {{kNoScriptId, kWebStoreNoScriptId},
+         {kUBlockId, kWebStoreUBlockId},
+         {kUMatrixId, kWebStoreUMatrixId},
+         {kAdGuardId, kWebStoreAdGuardId}});
+
+inline constexpr auto kWebStoreHosted =
+    base::MakeFixedFlatMap<std::string_view, std::string_view>(
+        {{kWebStoreNoScriptId, kNoScriptId},
+         {kWebStoreUBlockId, kUBlockId},
+         {kWebStoreUMatrixId, kUMatrixId},
+         {kWebStoreAdGuardId, kAdGuardId}});
+
+// In future there can be more brave-hosted mv2 extensions than published on
+// WebStore.
+static_assert(kBraveHosted.size() >= kWebStoreHosted.size());
+
+namespace internal {
+consteval std::array<std::string_view, kBraveHosted.size()>
+GetPreconfiguredManifestV2Extensions() {
+  // This can be made more idiomatic once Chromium style allows
+  // std::views::keys
+  std::array<std::string_view, kBraveHosted.size()> result{};
+  std::ranges::transform(kBraveHosted, result.begin(),
+                         [](const auto& p) { return p.first; });
+  return result;
+}
+}  // namespace internal
+
+inline constexpr auto kPreconfiguredManifestV2Extensions =
```

Comment:
it looked weird last time and it still looks weird this time.

this `constexpr` array is already based on other `constexpr` data baked into binary, and is used exactly once, so it can be created at runtime just fine.